packages/cashier/src/components/cashier-search-box/cashier-search-box.jsx

```diff
@@ -0,0 +1,74 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Field as FormField, Formik, Form } from 'formik';
+import { Icon, Input } from '@deriv/components';
+import './cashier-search-box.scss';
+
+const CashierSearchBox = ({ className, onClear, onSearch, placeholder, setIsSearchLoading }) => {
+    const onSearchClear = setFieldValue => {
+        setFieldValue('search', '');
+        if (typeof onClear === 'function') {
+            onClear();
+        }
+    };
+
+    const onSearchKeyUpDown = (submitForm, search) => {
+        if (!search.trim()) return;
+        setIsSearchLoading(true);
+        clearTimeout(typing_timer);
```

Comment:
after some investigations i decided to remove setTimeout because of it redundancy, because debounce function is already do this work, no matter how fast we enter the characters, the search will executed after 500 ms, it also helps me get rid of the flickering changing list of payment agents if the user presses and does not release the key on the keyboard